syntax = "proto3";

message BuildRequest {
  int32 id = 1;
  string namespace = 2;
  string name = 3;
  string framework = 4;
  string version = 5;
  string pyversion = 6;
  string resource = 7;
  string entry = 8;
  int32 revision = 9;
  string source = 10;
  repeated BuildIgnores ignores = 11;
  repeated BuildTags tags = 12;
}

message BuildIgnores {
  string value = 1;
}

message BuildTags {
	string name = 1;
	string value = 2;
}

message BuildResponse {
	int32 id = 1;
	string image = 2;
	string ecr_name = 3;
}

service ImageBuilder {
  // Simple RPC service for building images
	rpc Build(BuildRequest) returns (BuildResponse);
}